src/main/java/org/osc/manager/ism/entities/SecurityGroupEntity.java

```diff
@@ -31,52 +34,63 @@
 @Table(name = "SECURITY_GROUP")
 public class SecurityGroupEntity implements ManagerSecurityGroupElement {
 
-	@Id
-	@GeneratedValue
-	@Column(name = "id")
-	private Long id;
+    @Id
+    @GeneratedValue
+    @Column(name = "id")
+    private Long id;
 
-	@Column(name = "name", unique = true, nullable = false)
-	private String name;
+    @Column(name = "name", unique = true, nullable = false)
+    private String name;
 
-	@OneToOne(mappedBy = "securityGroup", fetch = FetchType.EAGER, cascade = CascadeType.ALL)
-	private SecurityGroupInterfaceEntity securityGroupInterfaces;
+    @OneToOne(mappedBy = "securityGroup", fetch = FetchType.EAGER, cascade = CascadeType.ALL)
+    private SecurityGroupInterfaceEntity securityGroupInterfaces;
 
-	public SecurityGroupEntity() {
-	}
+    @ManyToOne(fetch = FetchType.EAGER)
+    @JoinColumn(name = "device_sg_fk", nullable = false, foreignKey = @ForeignKey(name = "FK_SG_DEVICE"))
+    private DeviceEntity sgDevice;
 
-	public SecurityGroupEntity(String name) {
-		this.name = name;
-	}
+    public SecurityGroupEntity() {
+    }
 
-	public Long getId() {
-		return this.id;
-	}
+    public SecurityGroupEntity(String name) {
+        this.name = name;
+    }
 
-	public void setId(Long id) {
-		this.id = id;
-	}
+    public Long getId() {
+        return this.id;
+    }
 
-	public SecurityGroupInterfaceEntity getSecurityGroupInterfaces() {
-		return this.securityGroupInterfaces;
-	}
+    public void setId(Long id) {
+        this.id = id;
+    }
 
-	public void setSecurityGroupInterfaces(SecurityGroupInterfaceEntity securityGroupInterfaces) {
-		this.securityGroupInterfaces = securityGroupInterfaces;
-	}
+    public SecurityGroupInterfaceEntity getSecurityGroupInterfaces() {
+        return this.securityGroupInterfaces;
+    }
 
-	@Override
-	public String getName() {
-		return this.name;
-	}
+    public void setSecurityGroupInterfaces(SecurityGroupInterfaceEntity securityGroupInterfaces) {
+        this.securityGroupInterfaces = securityGroupInterfaces;
+    }
 
-	public void setName(String name) {
-		this.name = name;
-	}
+    @Override
+    public String getName() {
+        return this.name;
+    }
 
-	@Override
-	public String getSGId() {
-		return getId() == null ? null : getId().toString();
-	}
+    public void setName(String name) {
+        this.name = name;
+    }
 
+    @Override
+    public String getSGId() {
+        return getId() == null ? null : getId().toString();
+    }
+
+    public DeviceEntity getSgDevice() {
```

Comment:
getDevice, setDevice. The same comments for the SGIEntity

Reply:
same as above.